lib/paranoia.rb

```diff
@@ -249,6 +249,10 @@ def self.I_AM_THE_DESTROYER!
   def self.paranoid? ; false ; end
   def paranoid? ; self.class.paranoid? ; end
 
+  def self.delete_all(conditions=nil)
+    super and return unless paranoid?
```

Comment:
This is a bug. Now Model.delete_all always returns nil but not a number. It should be `return super unless paranoid?`.